### PR TITLE
Update TestContext.Properties return type to IDictionary<string, object?>

### PR DESCRIPTION
**Pull request by engyebrahim**
fix: https://github.com/microsoft/testfx/issues/1377

### CODE REVIEW

File: src/Adapter/MSTestAdapter.PlatformServices/Services/TestContextImplementation.cs [hunk 1]
```diff
@@ -116,7 +116,7 @@ public TestContextImplementation(ITestMethod testMethod, StringWriter stringWrit
 #endif
 
     /// <inheritdoc/>
-    public override IDictionary Properties => (IDictionary)_properties;
+    public override Dictionary<string, object?> Properties => _properties;
```
Comment (Evangelink) [inline src/Adapter/MSTestAdapter.PlatformServices/Services/TestContextImplementation.cs]: I would get rid of the field and replace this property with a readonly property.

**Review by Evangelink — COMMENTED** [review]
You can simplify furthermore the `GetProperty` method of `TestContext`.

@MarcoRossignoli Do you think we should go with `IDictionary` over `Dictionary`?

**Comment by MarcoRossignoli** [comment]
> @MarcoRossignoli Do you think we should go with IDictionary over Dictionary?

I would use always abstraction if possible so yes.

File: src/TestFramework/TestFramework.Extensions/TestContext.cs [hunk 2]
```diff
@@ -197,7 +197,7 @@ public abstract class TestContext
     {
         DebugEx.Assert(Properties is not null, "Properties is null");
 #if WINDOWS_UWP || WIN_UI
-        if (!((IDictionary<string, object>)Properties).TryGetValue(name, out object? propertyValue))
+        if (!Properties.TryGetValue(name, out object? propertyValue))
```
Comment (Evangelink) [inline src/TestFramework/TestFramework.Extensions/TestContext.cs:199]: You can merge all the #IF/#else part to keep only the `TryGetValue` part.

**Review by Evangelink — APPROVED** [review]

**Pull request merged**